File: src/leetcode/Problem1.java
package leetcode;

import java.util.HashMap;

class Problem1 {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++) {
            if(map.get(target-nums[i]) != null) {
                return new int[]{map.get(target-nums[i]), i};
            }
            map.put(nums[i], i);
        }
        return null;
    }

    public static void main(String[] args) {
        Problem1 solution = new Problem1();
        int[] rst = solution.twoSum(new int[]{2, 7, 11, 15}, 9);
        for (int e : rst)
            System.out.println(e);
    }
}

File: src/leetcode2nd/Problem1.cpp
#include <vector>
#include <map>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> m;
        vector<int> rt;
        for (int i = 0; i < nums.size(); i++) {
            if (m.find(target - nums.at(i)) != m.end()) {
                rt.push_back(m[target - nums.at(i)]);
                rt.push_back(i);
                return rt;
            }
            m[nums.at(i)] = i;
        }
        return rt;
    }
};